docs/normalize.md

```diff
@@ -0,0 +1,66 @@
+# The `normalize()` function
+
+Each base class defined using the NOMAD schema has a set of public functions which can be used at any moment when reading and parsing files in NOMAD (see their corresponding documentation pages). The only special situation happens for the `normalize(archive, logger)` function.
+
+This function is ran within the NOMAD infrastructure by the [`MetainfoNormalizer`](https://github.com/nomad-coe/nomad/blob/develop/nomad/normalizing/metainfo.py) in the following order:
+
+1. The deepest child class `normalize()` function is ran before their parents' `normalize()` function.
+2. For sibling sections, the `normalize()` function is executed from the smaller to the larger `normalizer_level` attribute. If `normalizer_level` is not set up, the order is established by the attributes definition order in the parent class.
+3. Using `super().normalize(archive, logger)` runs the inherited class normalize function.
+
+Let's see some examples. Imagine having the following `Section` and `SubSection` structure:
+
+```python
+from nomad.datamodel.data import ArchiveSection
+
+
+class Section1(ArchiveSection):
+    normalizer_level = 1
+
+    def normalize(self, achive, logger):
+        # some operations here
+        pass
+
+
+class Section2(ArchiveSection):
+    normalizer_level = 0
+
+    def normalize(self, achive, logger):
+        super().normalize(archive, logger)
+        # Some operations here or before `super().normalize(archive, logger)`
+
+
+class ParentSection(ArchiveSection):
+
+    sub_section_1 = SubSection(Section1.m_def, repeats=False)
+
+    sub_section_2 = SubSection(Section2.m_def, repeats=True)
+
+    def normalize(self, achive, logger):
+        super().normalize(archive, logger)
+        # Some operations here or before `super().normalize(archive, logger)`
+```
+
+Now, `MetainfoNormalizer` will be ran on the `ParentSection`. Applying **rule 1**, the `normalize()` functions of the `ParentSection`'s childs are executed first. The order of these functions is established by **rule 2** with the `normalizer_level` atrribute, i.e., all the `Section2` (note that `sub_section_2` is a list of sections) `normalize()` functions are ran first, then `Section1.normalize()`. Then, the order of execution will be:
```

Comment:
"will be run" and "functions are run first" (The 2nd one is tricky, but I think that is correct)